src/FileSystem-Memory/MemoryStore.class.st

```diff
@@ -7,24 +7,49 @@ Class {
 	#instVars : [
 		'root'
 	],
+	#classVars : [
+		'CurrentFS'
+	],
 	#category : #'FileSystem-Memory-Store'
 }
 
+{ #category : #current }
+MemoryStore class >> currentFileSystem [
+	^ CurrentFS ifNil: [ CurrentFS := FileSystem store: MemoryStore new ]
+]
+
 { #category : #public }
 MemoryStore class >> delimiter [
 	^ $/
 ]
 
+{ #category : #'class initialization' }
+MemoryStore class >> initialize [
+	"self initialize"
+
+	SessionManager default registerSystemClassNamed: self name
+]
+
 { #category : #public }
 MemoryStore class >> isCaseSensitive [
 	^ true
 ]
 
+{ #category : #'class initialization' }
+MemoryStore class >> reset [
+	CurrentFS := nil
+]
+
 { #category : #public }
 MemoryStore class >> separator [ 
 	^ $:
 ]
 
+{ #category : #'system startup' }
+MemoryStore class >> startUp [
+	self reset
+]
```

Comment:
This is strange... Why having a singleton if we are going to reset it every time we save the image?